### PR TITLE
Add 003-hello-segmentation notebook

### PR DESCRIPTION
**Pull request by Debskij**

### CODE REVIEW

**Review by ryanloney — CHANGES_REQUESTED** [review]
Looks good. Can you make the final visualizations bigger? I can hardly see it on my screen. 
image


Also, we should use one of the mapillary images that is from the vantage point of a car dashboard. I tried one of the CityScapes images and it works pretty well: 

image

File: notebooks/003-hello-segmentation/003-hello-segmentation.ipynb
Comment (helena-intel) [inline notebooks/003-hello-segmentation/003-hello-segmentation.ipynb:132]: To me this is confusing, we created input_image by transposing the original image, we now want the regular image back, we can just use `image` here?

File: notebooks/003-hello-segmentation/003-hello-segmentation.ipynb
Comment (helena-intel) [inline notebooks/003-hello-segmentation/003-hello-segmentation.ipynb:154]: FYI: we use American English spelling, so "visualize" etc. We will add this to the contribution guide.

File: notebooks/003-hello-segmentation/003-hello-segmentation.ipynb
Comment (helena-intel) [inline notebooks/003-hello-segmentation/003-hello-segmentation.ipynb:125]:
```suggestion
    "# Define the transparency of the segmentation mask on the photo\n",
```

File: notebooks/003-hello-segmentation/003-hello-segmentation.ipynb
Comment (helena-intel) [inline notebooks/003-hello-segmentation/003-hello-segmentation.ipynb:128]: The function from notebook_utils does not create the mask - the mask is created by argmax-ing the network output. The segmentation_map_to_image function transforms this mask, consisting of integer values from 0 to num_classes-1, to an RGB image with defined colors.

File: notebooks/003-hello-segmentation/003-hello-segmentation.ipynb
Comment (helena-intel) [inline notebooks/003-hello-segmentation/003-hello-segmentation.ipynb:122]:
```suggestion
    "# Define colormap, each color represents a class\n",
```

File: notebooks/003-hello-segmentation/003-hello-segmentation.ipynb
Comment (helena-intel) [inline notebooks/003-hello-segmentation/003-hello-segmentation.ipynb:10]: ADAS is not a very well known term so we could add a sentence explaining what it is https://www.adasandme.com/news/adas-is-now-adaptive-driver-assistance-systems/
and also have a sentence that says that this model recognizes four classes: background, road, curb and mark. 

```suggestion
    "We use the [road-segmentation-adas-0001](https://docs.openvinotoolkit.org/latest/omz_models_model_road_segmentation_adas_0001.html) model from [Open Model Zoo](https://github.com/openvinotoolkit/open_model_zoo/).\n"
```

Comment (helena-intel) [reply notebooks/003-hello-segmentation/003-hello-segmentation.ipynb:10]: When you search for OMZ models sometimes outdated documents are the first in the search results (you see that in the 2019_R1 in your link). We should link to the most recent version.

**Review by helena-intel — CHANGES_REQUESTED** [review]
Thanks @Debskij ! This is a welcome tutorial. I have a few change requests, look forward to merging this soon.

File: notebooks/003-hello-segmentation/003-hello-segmentation.ipynb
Comment (Debskij) [reply notebooks/003-hello-segmentation/003-hello-segmentation.ipynb]: Applied suggestions and added some description. Can you check is this what you ment?

File: notebooks/003-hello-segmentation/003-hello-segmentation.ipynb
Comment (Debskij) [reply notebooks/003-hello-segmentation/003-hello-segmentation.ipynb]: Changed comments, please check if this is fine now.

File: notebooks/003-hello-segmentation/003-hello-segmentation.ipynb
Comment (Debskij) [reply notebooks/003-hello-segmentation/003-hello-segmentation.ipynb]: Renamed variable higher, splitted input_image into resized_image and input_image, no need to use this input_image_t anymore.

File: notebooks/003-hello-segmentation/003-hello-segmentation.ipynb
Comment (Debskij) [reply notebooks/003-hello-segmentation/003-hello-segmentation.ipynb]: Changed comment and input function, please check if this is fine now

File: notebooks/003-hello-segmentation/003-hello-segmentation.ipynb
Comment (Debskij) [reply notebooks/003-hello-segmentation/003-hello-segmentation.ipynb]: > Looks good. Can you make the final visualizations bigger? I can hardly see it on my screen.
> image
> 
> Also, we should use one of the mapillary images that is from the vantage point of a car dashboard. I tried one of the CityScapes images and it works pretty well:
> 
> image

Used different image and resized output

**Comment by ryanloney** [comment]
The visualization looks much better now. Thank you! 
image

File: notebooks/003-hello-segmentation/003-hello-segmentation.ipynb
Comment (helena-intel) [inline notebooks/003-hello-segmentation/003-hello-segmentation.ipynb:72]:
```suggestion
    "image = cv2.imread(\"data/empty_road_mapillary.jpg\")\n",
```

`cv2.imread` by default loads the image as BGR, which is expected by the model. But with that, visualization will look off, because matplotlib expects RGB, so then you can created an `rgb_image` with `cv2.cvtColor(image, cv2.COLOR_BGR2RGB)` for visualization. See https://python.plainenglish.io/opencv-python-tutorial-for-beginners-part-1-7639496e98a

File: notebooks/003-hello-segmentation/003-hello-segmentation.ipynb
Comment (helena-intel) [inline notebooks/003-hello-segmentation/003-hello-segmentation.ipynb:148]: Can you switch around the base photo and the segmentation so the photo is left, then the mask, then the image?

**Review by helena-intel — CHANGES_REQUESTED** [review]

File: notebooks/003-hello-segmentation/003-hello-segmentation.ipynb
Comment (ryanloney) [reply notebooks/003-hello-segmentation/003-hello-segmentation.ipynb:72]: Thanks Helena. I was wondering why the color was off.

Comment (Debskij) [reply notebooks/003-hello-segmentation/003-hello-segmentation.ipynb:72]: Added one more variable that holds RGB picture as an example how to convert it back to format that can be displayed using matplotlib. Hope that is fine

File: notebooks/003-hello-segmentation/003-hello-segmentation.ipynb
Comment (helena-intel) [inline notebooks/003-hello-segmentation/003-hello-segmentation.ipynb:85]: This works well here, because the network aspect ratio is similar to the image aspect ratio. But that is not always the case - you may have images of 1024x768 and a network input shape of 512x512. Showing the resized image looks strange then. So for the general case, it is often useful to show the image in the original size and resize the mask to the original image size.

**Review by helena-intel — COMMENTED** [review]
Two minor comments - ready to merge after that. Thank you!

File: notebooks/003-hello-segmentation/003-hello-segmentation.ipynb
Comment (Debskij) [reply notebooks/003-hello-segmentation/003-hello-segmentation.ipynb]: Created rgb_image instead of RGB_resized_image, now it is rgb copy of base image. Other operation I added is resizing mask to meet H and W for base image.

**Comment by helena-intel** [comment]
Thanks @Debskij ! Also for your patience.  I saw that @ryanloney also requested changes - I set this to auto-merge so that it merges after he approves the PR.

**Review by ryanloney — APPROVED** [review]

**Pull request merged**
